libctru/source/services/mvd.c

```diff
@@ -304,10 +322,45 @@ void mvdstdExit(void)
 	MVDSTD_Shutdown();
 
 	svcCloseHandle(mvdstdHandle);
+	mvdstdHandle = 0;
 
 	linearFree(mvdstd_workbuf);
 }
 
+Result mvdstdCalculateBufferSize(MVDSTD_CalculateWorkBufSizeConfig* config, u32* size_out)
+{
+	bool must_close_handle = false;
+	Result ret = 0;
+
+	if(!config || !size_out)
+		return -1;
+
+	//If we don't have mvdstd handle, get it.
+	if(mvdstdHandle == 0)
+	{
+		if(R_FAILED(ret=srvGetServiceHandle(&mvdstdHandle, "mvd:STD")))
+			return ret;
+
+		must_close_handle = true;
+	}
+
+	//Attempting to use more than 0x10 for H.264 level will likely cause out-of-bound access. 
+	//https://www.3dbrew.org/wiki/MVDSTD:CalculateWorkBufSize
+	if(config->level.level > 0x10)
+		config->level.level = 0x10;
```

Comment:
This should be changed so that it returns an error instead of overwriting an input struct (& document that in the header)